### PR TITLE
[FIX](agg) fix vertical_compaction_reader for agg table with array/map type

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes
 fix vertical_compaction_reader for agg table with array/map type
same with this pr: https://github.com/apache/doris/pull/24873
Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

File: be/src/vec/olap/vertical_block_reader.cpp [hunk 1]
```diff
@@ -198,13 +198,23 @@ void VerticalBlockReader::_init_agg_state(const ReaderParams& read_params) {
...
-        // calculate `has_string` tag.
-        _stored_has_string_tag[idx] =
+        // calculate `_has_variable_length_tag` tag. like string, array, map
+        _stored_has_variable_length_tag[idx] =
                 _stored_data_columns[idx]->is_column_string() ||
                 (_stored_data_columns[idx]->is_nullable() &&
                  reinterpret_cast<ColumnNullable*>(_stored_data_columns[idx].get())
                          ->get_nested_column_ptr()
-                         ->is_column_string());
+                         ->is_column_string()) ||
```
Comment (xiaokang) [inline be/src/vec/olap/vertical_block_reader.cpp:207]: we should add a function is_variable_length() to IColumn and return true in ColumnString/Array/Map and any other variable length column.

Comment (xiaokang) [reply be/src/vec/olap/vertical_block_reader.cpp:207]: ColumnNullable call nested.is_variable_length()

**Comment by amorynan** [comment]
run buildall

**Review by xiaokang — DISMISSED** [review]
LGTM

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**